src/github.com/couchbase/sync_gateway/db/kv_change_index.go

```diff
@@ -190,37 +175,120 @@ func (k *kvChangeIndex) Stop() {
 	}
 }
 
-// Returns the stable sequence for a document from the stable clock, based on the vbucket for the document.
-// Used during document write for handling deduplicated sequences on the DCP feed.
-func (k *kvChangeIndex) GetStableSequence(docID string) SequenceID {
+// Returns the stable sequence for a document's vbucket from the stable clock.
+// Used during document write for handling deduplicated sequences on the DCP feed,
+// to determine when recent sequences can be pruned.
+// Uses reader's stable sequence to avoid db lookups, since document write doesn't
+// need the absolute latest version for pruning.
+func (k *kvChangeIndex) GetStableSequence(docID string) (seq SequenceID) {
 
 	vbNo := k.indexReadBucket.VBHash(docID)
-	return SequenceID{Seq: k.stableSequence.GetSequence(uint16(vbNo))}
-
-}
-
-// Returns the cached stable sequence clock for the index
-func (k *kvChangeIndex) getStableClock() base.SequenceClock {
-	return k.stableSequence.Clock
+	stableSequence, err := k.getReaderStableSequence()
+	if err != nil {
+		return seq
+	}
+	return SequenceID{Seq: stableSequence.AsClock().GetSequence(uint16(vbNo))}
 }
 
+// Used for external retrieval of the stable clock by cbgt.
 func (k *kvChangeIndex) GetStableClock() (clock base.SequenceClock, err error) {
+	// TODO: Should:
```

Comment:
Cleaned this up.  GetStableClock now actually does the described partition/counter check.  Removed the TODO, and moved the comment up.